code/cfile/cfile.h

```diff
@@ -462,4 +463,17 @@ namespace cfile
 
 }
 
+// This allows to use std::unique_ptr with CFILE
+namespace std {
+template <>
+struct default_delete<CFILE> {
```

Comment:
What do you think about adding a deleter to the unique_ptr as its second template argument?
That makes the intend clearer as the fact that the unique pointer closes the file and doesn't free memory is visible at points of use and not hidden in this template specialization.

Reply:
That would require using the correct template types for all `std::unique_ptr<CFILE>` usages which could be a problem and would also not be easily detectable since the code would compile but it would fail once the `unique_ptr` was deleted.

With this, it is always valid to use `std::unique_ptr<CFILE>`.

Reply:
I personally wouldn't expect a `unique_ptr` to do more than delete the memory resource when I don't see a custom deleter.

I'm not sure if your deleter wouldn't leak the CFILE, if someone for whatever reasons allocates one on the heap and puts it into the `unique_ptr`.

Maybe the safest solution is a `CfileHandle` struct, which does the clean-up in its destructor and is move only, then memory management and possible CFILEs on the heap are separated from file clean-up.

Reply:
`CFILE` is an incomplete type so no one could allocate one on the heap without hacking into the internals of the CFile system.

Also, this code doesn't break the assumption that `unique_ptr` simply deallocates memory (which isn't even universally true) because it actually does "deallocate" memory by passing the pointer to `cfclose`.

Reply:
Isn't CFILE defined in the cfile.h, public header? Maybe its definition could be moved to the .cpp

If it is an incomplete type the generic default deleter and thus unique pointer would not compile.

I'm not against the changeset in general. I just think putting the behaviour into a custom deleter makes intent clearer, but I admit that in the end its more a matter of personal preference.

Reply:
Huh, I must have misremembered that somehow.

Anyway, a valid `CFILE` pointer can only be created by the cfile system so using `cfclose` is the correct way for deleting that so I think using the default deleter way is the correct solution here.